src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -326,11 +496,25 @@ impl<'a, T: Timestamp + Lattice> StorageControllerMut<'a, T> {
             if !change.is_empty() {
                 let frontier = self
                     .as_ref()
-                    .collection(*id)
-                    .unwrap()
+                    .collection(*id)?
                     .read_capabilities
                     .frontier()
                     .to_owned();
+
+                let mut ts_binding_stash = self
+                    .storage
+                    .stash
+                    .collection::<PartitionId, ()>(&id.to_string())?;
+
+                let mut since = ts_binding_stash.since()?;
+                since.extend(
+                    frontier
+                        .iter()
+                        .map(|t| t.clone().try_into().expect("timestamp overflowed i64")),
+                );
+                ts_binding_stash.compact(since.borrow())?;
+                ts_binding_stash.consolidate()?;
```

Comment:
Consolidating on every update might be expensive. Not sure! It matches what we currently do, perhaps, and so is maybe fine?